python-frontend/src/main/java/org/sonar/python/semantic/v2/TypeInferenceV2.java

```diff
@@ -147,7 +154,7 @@ private void flowSensitiveTypeInference(ControlFlowGraph cfg, Set<SymbolV2> trac
       parameterTypes);
 
     flowSensitiveTypeInference.compute(cfg);
-    flowSensitiveTypeInference.compute(cfg);
+    return ((TypeInferenceProgramState) flowSensitiveTypeInference.compute(cfg)).typesBySymbol();
```

Comment:
I'd get rid of this class cast here by overloading the `compute` method for the `FlowSensitiveTypeInference`
```java
public class FlowSensitiveTypeInference extends ForwardAnalysis {
  ...
  @Override
  public TypeInferenceProgramState compute(ControlFlowGraph cfg) {
    return (TypeInferenceProgramState) super.compute(cfg);
  }
  ...
```